### DISCUSSION TITLE
support advanced clickable divs

### DISCUSSION THREAD

**Discussion by hpvd** [unpoly/unpoly#535] [Ideas]
"cards" or even "rows" in tables are often used as entry points/links on overview pages/categories to deeper information or subinteractions
This is not as easy it sounds since these cards/rows often contain several elements, it should be fine for seo and accessibility, they should have hover effects for good orientation etc
see as examples
- cards (not great as example): https://unpoly.com/
- rows: https://demo.unpoly.com/projects

so there are several requirements:
1. The whole thing is linked and clickable.
2. It is able to contain more than one link.
3. This content is semantic so assistive tech can understand it.
4. The text should be selectable, like regular links.
5. Things like right-click and keyboard shortcuts should work with it
6. Its elements should be focusable when tabbing.

As learned this could be solved this way:
https://codepen.io/vikas-parashar/pen/qBOwMWj

![2023-09-21_18h27_35](https://github.com/unpoly/unpoly/assets/5681880/f665feba-43c4-4cd9-99f7-d5fa46fc60a5)

source and description:
https://css-tricks.com/block-links-the-search-for-a-perfect-solution/#method-4-sprinkle-javascript-on-the-second-method

=> maybe it does make sense to take this topic into account for unpoly and
provide a POC in Demo?

could be combined with 
open and close overlay to start point 
https://github.com/unpoly/unpoly/discussions/534
to make it even more advanced...

**Comment by alexandremjacques** [comment]
My guess is that this shouldn't be of Unpoly concern. AFAIK, Unpoly is pretty compliant to HTML Spec.

You could make use of Unpoly compilers (https://unpoly.com/up.compiler) and `up.on` (https://unpoly.com/up.on) to achive what you want.

**Reply by hpvd** [reply]
many thanks for pointing to this!

**Comment by triskweline** [comment]
You can add an [`[up-expand]`](https://unpoly.com/up-expand) attribute to any element to enlarge the click area of a descendant link.

**Reply by hpvd** [reply]
thank you!
